File: exe2shell.py
import sys

def read_file_in_chunks(file_location, chunk_size=8192):
    try:
        content = bytearray()
        with open(file_location, 'rb') as file:
            while (chunk := file.read(chunk_size)):
                content.extend(chunk)
        return content
    except:
        return None

def format_content(content):
    if content is not None:
        return '\\x' + '\\x'.join(format(byte, '02x') for byte in content)
    else:
        return None

def main():
    if len(sys.argv) != 2:
        print("Please provide a file name as an argument.")
        return
    file_location = sys.argv[1]
    content = read_file_in_chunks(file_location)
    formatted_content = format_content(content)
    if formatted_content is not None:
        print(formatted_content)

if __name__ == "__main__":
    main()

File: exe2shell.rs
use std::env;
use std::fs::File;
use std::io::Read;

fn read_file(file_location: &str) -> Option<Vec<u8>> {
    let mut file = File::open(file_location).ok()?;
    let mut buffer = Vec::new();
    file.read_to_end(&mut buffer).ok()?;
    Some(buffer)
}

fn format_content(content: Option<Vec<u8>>) -> Option<String> {
    content.map(|bytes| bytes.iter().map(|b| format!("\\x{:02x}", b)).collect())
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 2 {
        eprintln!("Please provide a file name as an argument.");
        return;
    }
    let file_location = &args[1];
    let content = read_file(file_location);
    let formatted_content = format_content(content);
    if let Some(formatted_content) = formatted_content {
        println!("{}", formatted_content);
    }
}
